core/src/main/java/hudson/model/Items.java

```diff
@@ -79,6 +80,44 @@
             return false;
         }
     };
+    /**
+     * A comparator of {@link Item} instances that uses a case-insensitive comparison of {@link Item#getName()}.
```

Comment:
Wouldn't a natural sort be better? IIRC a user recently requested this.

I.e. 

    item1
    item2
    item10